synapse/config/tracer.py

```diff
@@ -24,41 +24,46 @@ class TracerConfig(Config):
     section = "tracing"
 
     def read_config(self, config: JsonDict, **kwargs: Any) -> None:
-        opentracing_config = config.get("opentracing")
-        if opentracing_config is None:
-            opentracing_config = {}
+        tracing_config = config.get("tracing")
+        if tracing_config is None:
+            tracing_config = {}
 
-        self.opentracer_enabled = opentracing_config.get("enabled", False)
+        self.tracing_enabled = tracing_config.get("enabled", False)
 
-        self.jaeger_config = opentracing_config.get(
-            "jaeger_config",
-            {"sampler": {"type": "const", "param": 1}, "logging": False},
+        self.jaeger_exporter_config = tracing_config.get(
+            "jaeger_exporter_config",
+            {},
         )
 
         self.force_tracing_for_users: Set[str] = set()
 
-        if not self.opentracer_enabled:
+        if not self.tracing_enabled:
             return
 
-        check_requirements("opentracing")
+        check_requirements("opentelemetry")
 
         # The tracer is enabled so sanitize the config
 
-        self.opentracer_whitelist: List[str] = opentracing_config.get(
+        # Default to always sample. Range: [0.0 - 1.0]
+        self.sample_rate: float = float(tracing_config.get("sample_rate", 1))
+        if self.sample_rate < 0.0 or self.sample_rate > 1.0:
+            raise ConfigError("Tracing sample_rate must be in range [0.0, 1.0].")
+
+        self.homeserver_whitelist: List[str] = tracing_config.get(
```

Comment:
While we're renaming all of the config here, do we have interest in an `allowlist` rename?